platform/darwin/src/MGLStyleLayer.h.ejs

```diff
@@ -0,0 +1,29 @@
+<%
+  const type = locals.type;
+  const layoutProperties = locals.layoutProperties;
+  const paintProperties = locals.paintProperties;
+-%>
+// This file is generated. Do not edit.
+
+#import <Mapbox/Mapbox.h>
+
+#import "MGLTypes.h"
+
+@interface MGL<%- camelize(type) %>StyleLayer : MGLStyleLayer
+
+<% if (layoutProperties.length) { -%>
+// Layout properties
+
+<% for (const property of layoutProperties) { -%>
+@property (nonatomic) <%- propertyType(property) %> <% if (isObject(property)) { -%>*<% } -%><%- camelizeWithLeadingLowercase(property.name) %>;
+
+<% } -%>
+<% } -%>
+// Paint properties
+
+<% for (const property of paintProperties) { -%>
```

Comment:
Putting paint properties at the top level of this object means we can only support default paint properties. Given that the iOS SDK [supports style classes](https://github.com/mapbox/mapbox-gl-native/blob/03ebae0f2ed0e5ff2c4fe284d8d3db8960ae4938/platform/ios/src/MGLMapView.h#L214) (and the macOS SDK could trivially do so as well), do we also need to support [class-specific paint properties](https://www.mapbox.com/mapbox-gl-style-spec/#layer-paint.*)?

/cc @jfirebaugh